### DISCUSSION TITLE
Qt highlight colour and progress bar dimensions

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#5872] [Q&A]
After experiencing the pitfalls of #5800 I'm trying to change the highlight colour in the Qt client using the current coding in TorrentDelegate.cc and TorrentDelegateMin.cc.

Generally, all of the other Transmission clients will use a blue highlight colour, whereas Qt currently uses the green seeding colour to highlight individual torrents.

I could be completely misreading the code, but, for example, the current code in TorrentDelegate.cc (similarly in TorrentDelegateMin.cc) that _looks_ like it changes these values is as follows.

`    if (tor.isDownloading())
    {
        progress_bar_style_.palette.setBrush(QPalette::Highlight, BlueBrush);
        progress_bar_style_.palette.setColor(QPalette::Base, BlueBack);
        progress_bar_style_.palette.setColor(QPalette::Window, BlueBack);
    }
    else if (tor.isSeeding())
    {
        progress_bar_style_.palette.setBrush(QPalette::Highlight, GreenBrush);
        progress_bar_style_.palette.setColor(QPalette::Base, GreenBack);
        progress_bar_style_.palette.setColor(QPalette::Window, GreenBack);
    }
    else
    {
        progress_bar_style_.palette.setBrush(QPalette::Highlight, GreenBrush);
        progress_bar_style_.palette.setColor(QPalette::Base, SilverBack);
        progress_bar_style_.palette.setColor(QPalette::Window, SilverBack);`

If i make the following changes, nothing changes when i rebuild.

`    if (tor.isDownloading())
    {
        progress_bar_style_.palette.setBrush(QPalette::Highlight, BlueBrush);
        progress_bar_style_.palette.setColor(QPalette::Base, BlueBack);
        progress_bar_style_.palette.setColor(QPalette::Window, BlueBack);
    }
    else if (tor.isSeeding())
    {
        progress_bar_style_.palette.setBrush(QPalette::Highlight, BlueBrush);
        progress_bar_style_.palette.setColor(QPalette::Base, GreenBack);
        progress_bar_style_.palette.setColor(QPalette::Window, GreenBack);
    }
    else
    {
        progress_bar_style_.palette.setBrush(QPalette::Highlight, BlueBrush);
        progress_bar_style_.palette.setColor(QPalette::Base, SilverBack);
        progress_bar_style_.palette.setColor(QPalette::Window, SilverBack);`

Some time this year the Qt client changed its progress bar dimensions to something far more slim from this.

![qt-ubuntu](https://user-images.githubusercontent.com/69029666/208224940-dc7008b8-f21c-49a5-b205-ebe80c1b666f.png)

Playing around with different values for the progress bar in TorrentDelegateMin.cc (Lines 32-35) doesn't seem to change anything either. Any help?

**Comment by stefanos82** [comment]
I guess it's a desktop theme thing, because I use XFCE 4 and this is what my `transmission-qt` looks like:

![image](https://github.com/transmission/transmission/assets/1777466/3974a081-9ab8-4af9-9c60-de8be21b56d9)
